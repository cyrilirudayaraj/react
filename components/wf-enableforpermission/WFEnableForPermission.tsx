import React, { Component } from 'react';
import AuthUtil from '../../utils/AuthUtil';
import { getPermissions } from '../../slices/UserPermissionSlice';
import { connect } from 'react-redux';

interface WFEnableForPermissionProps {
  permission?: any;
  userPermissions?: any;
  children?: any;
}

class WFEnableForPermission extends Component<WFEnableForPermissionProps> {
  render() {
    const { permission, children } = this.props;
    const isDisabled = !AuthUtil.isAuthorized(
      permission,
      this.props.userPermissions
    );
    if (isDisabled) {
      const newChildren = React.Children.map(children, function (child: any) {
        return React.cloneElement(child, { disabled: isDisabled });
      });
      return newChildren;
    }
    return children;
  }
}

const mapStateToProps = (state: any) => {
  return { userPermissions: getPermissions(state) };
};

export default connect(mapStateToProps)(WFEnableForPermission);
