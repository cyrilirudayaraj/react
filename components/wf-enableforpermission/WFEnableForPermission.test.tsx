import React from 'react';
import { configure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import WFEnableForPermission from './WFEnableForPermission';
import { Button } from '@athena/forge';

configure({ adapter: new Adapter() });
let wrapper: any;
let store: any;
beforeEach(() => {
  window['globalVars'] = {
    enablePermission: 1,
  };
  store = configureStore([])({
    userPermission: {
      userPermissions: ['task.update'],
    },
  });
  store.dispatch = jest.fn();
});
describe('snapshot test <WFEnableForPermission>', () => {
  it('should render as expected with permission', () => {
    wrapper = shallow(
      <Provider store={store}>
        <WFEnableForPermission permission={'task.update'} />
      </Provider>
    );
    wrapper.setProps({ userPermissions: ['task.update'] });
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('should render the child component with permission', () => {
    wrapper = mount(
      <Provider store={store}>
        <WFEnableForPermission permission={'task.update'}>
          <Button text={'testButton'} />
        </WFEnableForPermission>
      </Provider>
    );
    expect(wrapper.find(Button).props().disabled).toEqual(false);
  });

  it('should disable the child component', () => {
    wrapper = mount(
      <Provider store={store}>
        <WFEnableForPermission permission={'task.create'}>
          <Button text={'testButton'} />
        </WFEnableForPermission>
      </Provider>
    );
    expect(wrapper.find(Button).props().disabled).toEqual(true);
  });
});
