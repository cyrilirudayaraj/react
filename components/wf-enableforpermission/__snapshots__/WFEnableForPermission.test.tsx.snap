// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test <WFEnableForPermission> should render as expected with permission 1`] = `
"<ContextProvider value={{...}}>
  <Connect(WFEnableForPermission) permission=\\"task.update\\" />
</ContextProvider>"
`;
